src/main/java/com/learning/mfscreener/service/HistoricalNavService.java

```diff
@@ -96,27 +98,22 @@ String parseNavData(
                 String[] tokenize = lineValue.split(AppConstants.NAV_SEPARATOR);
```

Comment:
@coderabbitai, refactor this method to reduce cognitive complexity without loosing functionality.

Reply:
@coderabbitai, above code is not ignoring first two lines and when oldSchemeId is set, loop should be broken. Please adjust accordingly.